pkg/isoeditor/rhcos.go

```diff
@@ -68,8 +74,60 @@ func CreateMinimalISO(extractDir, volumeID, rootFSURL, arch, minimalISOPath stri
 	return nil
 }
 
+func execute(command, workDir string) (string, error) {
+	var stdoutBytes, stderrBytes bytes.Buffer
+	cmd := exec.Command("bash", "-c", command)
+	cmd.Stdout = &stdoutBytes
+	cmd.Stderr = &stderrBytes
+	log.Infof(fmt.Sprintf("Running cmd: %s\n", command))
+	cmd.Dir = workDir
+	err := cmd.Run()
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to execute cmd (%s): %s", cmd, stderrBytes.String())
+	}
+
+	return strings.TrimSuffix(stdoutBytes.String(), "\n"), nil
+}
+
+func extractNmstatectl(extractDir, workDir string) (string, error) {
+	nmstateDir, err := os.MkdirTemp(workDir, "nmstate")
+	if err != nil {
+		return "", err
+	}
+	rootfsPath := filepath.Join(extractDir, "images/pxeboot/rootfs.img")
+	_, err = execute(fmt.Sprintf("7z x %s", rootfsPath), nmstateDir)
+	if err != nil {
+		log.Errorf("failed to 7z x rootfs.img: %v", err.Error())
+		return "", err
+	}
+	// limiting files is needed on el<=9 due to https://github.com/plougher/squashfs-tools/issues/125
+	ulimit := "ulimit -n 1024"
```

Comment:
In case it helps: the `ulimit -n ...` builtin command only changes the limit for process of the shell that runs it, and for the processes created by that shell. As the code below only uses it before running `unsquashfs ...` it will have no effect in any other thing that uses this image.